cip/CIP2016-07-07-Parameter-syntax.adoc

```diff
@@ -0,0 +1,91 @@
+= CIP2016-07-07 - Parameter syntax
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Mats Rydberg <mats@neotechnology.com>
+
+[abstract]
+.Abstract
+--
+This CIP suggests changing syntax for parameter usage, from curly braces (`{}`) to dollars (`$`).
+Curly brackets are used by the language for literal maps and property predicates, and this overloading may cause confusion in usage.
+Using a different, unique character causes parameters to be more visible for the human eye, and by using only a single character for the parameter instead of one opening and one closing, the syntax becomes somewhat cleaner.
+--
+
+toc::[]
+
+
+== Motivation
+
+//Provide the rationale and justification for the proposed feature(s).
+
+The clause `MATCH (a {prop: {param}})` contains a nested curly brace structure, which is not ideal for human readability.
+The alternative `MATCH (a {prop: $param})` does a much better job at signalling the parameter usage to the reader.
+
+== Background
+
+This is further background regarding the CIP, should it be required or deemed useful.
+
+Please ensure that links to related documents (including other CIPs) are provided.
+
+== Proposal
+
+The concrete suggestion made by this CIP is to replace the parameter syntax with curly braces with a dollar sign.
```

Comment:
This sentence is very confusing; please replace 
'The concrete suggestion made by this CIP is to replace the parameter syntax with curly braces with a dollar sign.'
with (or something similar):
'This CIP proposes using the dollar sign instead of curly braces to delimit parameters.'